### PR TITLE
net: IPv6, accept_ra, slaac, stateless

### PR DESCRIPTION
**Pull request by hjensas**
Router advertisements are required for the default route
to be set up, thus accept_ra should be enabled for
dhcpv6-stateful.

sysconf: IPV6_FORCE_ACCEPT_RA controls accept_ra sysctl.
eni: mode static and mode dhcp 'accept_ra' controls sysctl.

Add 'accept-ra: true|false' parameter to config v1 and
v2. When True: accept_ra is set to '1'. When False:
accept_ra is set to '0'. When not defined in config the
value is left to the operating system default.

This change also extend the IPv6 support to distinguish
between slaac and dhcpv6-stateless. SLAAC is autoconfig
without any options from DHCP, while stateless auto-configures
the address and the uses DHCP for other options.

LP: #1806014
LP: #1808647

### CODE REVIEW

File: cloudinit/net/sysconfig.py [hunk 1]
```diff
@@ -466,10 +487,17 @@ def _render_subnet_routes(cls, iface_cfg, route_cfg, subnets):
...
+                    # default gateway come from RA's.
+                    # (https://github.com/openSUSE/wicked/issues/570)
                     if iface_cfg['BOOTPROTO'] in ('dhcp', 'dhcp4', 'dhcp6'):
+                        # NOTE(hjensas): DHCLIENT_SET_DEFAULT_ROUTE is SuSE
+                        # only. RHEL, CentOS, Fedora does not implement this
+                        # option.
                         iface_cfg['DHCLIENT_SET_DEFAULT_ROUTE'] = True
                     if 'gateway' in route:
-                        if is_ipv6 or is_ipv6_addr(route['gateway']):
+                        if is_ipv6:
```
Comment (raharper) [inline cloudinit/net/sysconfig.py]: Won't we miss in a purely static ipv6 configuration?

File: cloudinit/net/sysconfig.py [hunk 2]
```diff
@@ -444,10 +461,14 @@ def _render_subnets(cls, iface_cfg, subnets, has_default_route):
     @classmethod
     def _render_subnet_routes(cls, iface_cfg, route_cfg, subnets):
         for _, subnet in enumerate(subnets, start=len(iface_cfg.children)):
+            subnet_type = subnet.get('type')
             for route in subnet.get('routes', []):
                 is_ipv6 = subnet.get('ipv6') or is_ipv6_addr(route['gateway'])
```
Comment (hjensas) [inline cloudinit/net/sysconfig.py]: is_ipv6 is true if route['gateway'] is an ipv6 address here.

File: cloudinit/net/sysconfig.py (hunk 1 above)
Comment (hjensas) [reply cloudinit/net/sysconfig.py]: is_ipv6 is true if route['gateway'] is an IPv6 address, see my comment on L466 above.

File: cloudinit/net/sysconfig.py (hunk 2 above)
Comment (raharper) [reply cloudinit/net/sysconfig.py]: Ah, perfect.  Thanks!

**Comment by irvingpop** [comment]
Thank you @raharper !!!

**Comment by raharper** [comment]
Big thanks to @hjensas for putting it all together.

**Pull request merged**